src/copilot/copilotHelper.ts

```diff
@@ -0,0 +1,358 @@
+                        cancellationToken.onCancellationRequested(() => {
+                            reject(new Error(ErrorMessage.OperationCancelled));
+                        });
+                    }),
+                    new Promise<IProjectDependenciesResult>((_, reject) => {
+                        setTimeout(() => {
+                            reject(new Error(ErrorMessage.OperationTimedOut));
+                        }, 40); // 40ms timeout
+                    })
+                ]);
+                if (!result) {
+                    return {
+                        dependencyInfoList: [],
+                        emptyReason: EmptyReason.CommandNullResult,
+                        isEmpty: true
+                    };
+                }
+                return result;
+            } else {
+                const result = await Promise.race([
+                    commandPromise,
+                    new Promise<IProjectDependenciesResult>((_, reject) => {
+                        setTimeout(() => {
+                            reject(new Error(ErrorMessage.OperationTimedOut));
+                        }, 40); // 40ms timeout
+                    })
+                ]);
+                if (!result) {
+                    return {
+                        dependencyInfoList: [],
+                        emptyReason: EmptyReason.CommandNullResult,
+                        isEmpty: true
+                    };
+                }
+                return result;
+            }
+        } catch (error: any) {
+            if (error.message === ErrorMessage.OperationCancelled) {
+                return {
+                    dependencyInfoList: [],
+                    emptyReason: EmptyReason.CopilotCancelled,
+                    isEmpty: true
+                };
+            }
+            if (error.message === ErrorMessage.OperationTimedOut) {
+                return {
+                    dependencyInfoList: [],
+                    emptyReason: EmptyReason.Timeout,
+                    isEmpty: true
+                };
+            }
+            const errorMessage = 'TsException_' + ((error as Error).message || "unknown");
+            sendError(new GetProjectDependenciesError(errorMessage));
+            return {
+                dependencyInfoList: [],
+                emptyReason: errorMessage,
+                isEmpty: true
+            };
+        }
+    }
+
+    /**
+     * Resolves project dependencies and converts them to context items with cancellation support
+     * @param workspaceFolders The workspace folders, or undefined if none
+     * @param copilotCancel Cancellation token from Copilot
+     * @param checkCancellation Function to check for cancellation
+     * @returns Array of context items for project dependencies, or empty array if no workspace folders
+     */
+    export async function resolveAndConvertProjectDependencies(
+        workspaceFolders: readonly { uri: Uri }[] | undefined,
```

Comment:
A workspace folder is not necessarily mapped to a Java project. A better approach is to find the related Java project based on the active editor file, not based on the workspace folder info.

Reply:
yes, this is a good point! I also find that java project have aggregate project, so this PR is going to support such scenario
https://github.com/microsoft/vscode-java-dependency/pull/926
My consideration is that if, in the end, we can provide the entire workspace's project info, then we can consider updating them simultaneously.